File: 318. Maximum Product of Word Lengths/Python/Maximum Product of Word Lengths.py
class Solution(object):
    def maxProduct(self, words):
        """
        :type words: List[str]
        :rtype: int
        """
        bits = []
        result = 0
        a_in_ascii = ord('a')
        for word in words:
            bit = 0
            for letter in word:
                bit |= 1 << (ord(letter) - a_in_ascii)
            bits.append(bit)
        for i in range(len(bits)):
            for j in range(i + 1, len(bits)):
                if not bits[i] & bits[j] and len(words[i]) * len(words[j]) > result:
                    result = len(words[i]) * len(words[j])
        return result

File: 318. Maximum Product of Word Lengths/Java/Maximum Product of Word Lengths.java
public class Solution {
    public int maxProduct(String[] words) {
        int[] bits = new int[words.length];
        int result = 0;
        for (int i = 0; i < words.length; i++) {
            int bit = 0;
            for (char letter : words[i].toCharArray()) {
                bit |= 1 << (letter - 'a');
            }
            bits[i] = bit;
        }
        for (int i = 0; i < bits.length; i++) {
            for (int j = i + 1; j < bits.length; j++) {
                if ((bits[i] & bits[j]) == 0 && words[i].length() * words[j].length() > result) {
                    result = words[i].length() * words[j].length();
                }
            }
        }
        return result;
    }
}